ui/src/app/shared/header/header.component.ts

```diff
@@ -192,10 +192,16 @@ export class HeaderComponent implements OnInit, AfterViewInit, AfterViewChecked,
       }
     });
     let query: string = URLSearchParamsUtils.encodeURLSearchParamsFromMap(paramMap);
-    this.router.navigate(
-      ['jobs'],
-      {queryParams: { q: query}}
-    );
+    if (query) {
+      this.router.navigate(
+        ['jobs'],
+        {queryParams: { q: query}}
+      );
+    } else if (this.route.snapshot.queryParams['q']) {
```

Comment:
nit: I might just depend on the router to figure this out, i.e. always navigate but exclude "q" if unfiltered. There may be more edge cases which wouldn't be caught here, e.g.

- Navigate to laggy filtered page from unfiltered
- Change my mind and remove the filter while it's loading

In that situation, the router would probably cancel the first navigate.

Reply:
done